core/domain/user_services.py

```diff
@@ -2022,47 +2034,67 @@ def add_created_exploration_id(user_id: str, exploration_id: str) -> None:
     user_contributions = get_user_contributions(user_id, strict=False)
 
     if not user_contributions:
-        create_user_contributions(user_id, [exploration_id], [])
+        updated_user_contributions = compute_user_contributions(
+            user_id, [exploration_id], []
+        )
+        updated_user_contributions.update_timestamps()
+        updated_user_contributions.put()
     elif exploration_id not in user_contributions.created_exploration_ids:
         user_contributions.created_exploration_ids.append(exploration_id)
         user_contributions.created_exploration_ids.sort()
-        _save_user_contributions(user_contributions)
+        _get_validated_user_contributions_model(user_contributions).put()
 
 
-def add_edited_exploration_id(user_id: str, exploration_id: str) -> None:
+def add_edited_exploration_id(
+    user_id: str,
+    exploration_id: str
+) -> List[user_models.UserContributionsModel]:
     """Adds an exploration_id to a user_id's UserContributionsModel collection
     of edited explorations.
 
     Args:
         user_id: str. The unique ID of the user.
         exploration_id: str. The exploration id.
+
+    Returns:
+        UserContributionsModel. The updated UserContributionsModel.
     """
+    user_contribution_models = []
     user_contributions = get_user_contributions(user_id, strict=False)
 
     if not user_contributions:
-        create_user_contributions(user_id, [], [exploration_id])
+        user_contribution_models.append(
+            compute_user_contributions(user_id, [], [exploration_id])
+        )
 
     elif exploration_id not in user_contributions.edited_exploration_ids:
         user_contributions.edited_exploration_ids.append(exploration_id)
         user_contributions.edited_exploration_ids.sort()
-        _save_user_contributions(user_contributions)
+        user_contribution_models.append(
+            _get_validated_user_contributions_model(user_contributions)
+        )
+    return user_contribution_models
 
 
-def _save_user_contributions(
+def _get_validated_user_contributions_model(
     user_contributions: user_domain.UserContributions
-) -> None:
-    """Commits a user contributions object to the datastore.
+) -> user_models.UserContributionsModel:
+    """Constructs a valid UserContributionsModel.
```

Comment:
"Constructs a valid UserContributionsModel from the given domain object.

This function does not save anything to the datastore."